src/huggingface_hub/snapshot_download.py

```diff
@@ -41,8 +42,19 @@ def snapshot_download(
     if isinstance(cache_dir, Path):
         cache_dir = str(cache_dir)
 
+    if isinstance(use_auth_token, str):
+        token = use_auth_token
+    elif use_auth_token:
+        token = HfFolder.get_token()
+        if token is None:
+            raise EnvironmentError(
+                "You specified use_auth_token=True, but a Hugging Face token was not found."
+            )
```

Comment:
Not for this PR as it's unrelated, but maybe we should refactor this into a function (`get_token_or_raise`?) or add an optional param to error if no token. We have logic similar to this in multiple laces at the moment:

* https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/file_download.py#L331-L336
* https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/repository.py#L317
* + some other third party libraries for which a token is strictly required

Reply:
Yeah I agree this would be a nice improvement - happy to tackle it after this PR is merged (so I can make progress on SUPERB :))

Reply:
I agree as well!